### PR TITLE
#499 Calculate workshop balance automatically

### PR DESCRIPTION
**Pull request by bgeree**
Please check if the `WorkshopBalance::generateBalance()` method is called wherever it's needed.

### CODE REVIEW

File: app/Models/WorkshopBalance.php [hunk 1]
```diff
@@ -28,56 +25,64 @@ public function semester()
...
+        if(!self::where('semester_id', $semester_id)->count()) {
+            $balances = [];
+            foreach ($workshops as $workshop) {
+                $balances[] = [
+                    'semester_id' => $semester_id,
+                    'workshop_id' => $workshop->id
+                ];
+            }
+
+            self::insert($balances);
```
Comment (luksan47) [inline app/Models/WorkshopBalance.php:46]: This line fails with `Field 'resident' doesn't have a default value`.

**Review by luksan47 — CHANGES_REQUESTED** [review]
Hey, I'm really sorry we have not checked your PR, we got really distracted... :(

Run through the PR now, it looks good, thanks for working on it!
Found an issue, when trying to add payment to a user (see comment below). If you could fix this, it would be great, but I understand if you are not interested anymore.

File: app/Models/WorkshopBalance.php (hunk 1 above)
Comment (bgeree) [reply app/Models/WorkshopBalance.php:46]: I'll take a look at it this week!

Comment (bgeree) [reply app/Models/WorkshopBalance.php:46]: Sorry for being slow, I fixed the issue!

**Review by kdmnk — APPROVED** [review]
Thanks, seems fine to me, we will have to add test for this anyways (I consulted with the committe leader for the details.)

**Review by luksan47 — APPROVED** [review]
Thanks @bgeree for working on this! LGTM!

**Pull request merged**